### ISSUE TITLE
Run & Analyze gas measurmens on Arbitrum Nitro Goerli

### ISSUE THREAD

**Issue by jacque006** [jzaki/bls-wallet#489]
*As a* developer using BLS Wallet
*I would like to* understand the gas differences/savings/costs of using BLS Wallet on Arbitrum Nitro in common transactions
*So that I can* determine if it is the right fit for my projects needs

### Notes

- Produce a document which includes gas measurements & analysis of running gas measurements on Arbitrum Nitro Goerli for sharing with internal & external shareholders.

**Comment by voltrevo** [comment]
This feels a bit testnet focused. Can we include a note here that the real goal (as part of 'analyze'?) is to get an estimate of the costs on Arbitrum Nitro (for-realz)? (That is the real goal here, right?)

**Comment by jacque006** [comment]
@voltrevo see https://github.com/web3well/bls-wallet/issues/490 (& https://github.com/web3well/bls-wallet/milestone/9)

**Comment by voltrevo** [comment]
@jacque006 Right, but I just don't see the value in running this measurement on Arbitrum Goerli except as a proxy to estimate real Arbitrum. Maybe the value is in refining/practicing the process before spending the fees on mainnet?

But then, even when running on mainnet, the L1/L2 gas ratio fluctuates all the time, so I see this measurement as quite weak.

Similar to what we touched on in our call, the (L1 gas, L2 gas) pair is the thing that we can control. If we don't have those two numbers (remembering that "gas used for L1" is *not* L1 gas), then the measurement we have will be subject to the constant gas variability that is outside of our project.

To put it another way, if we have the (L1 gas, L2 gas) pair, then we could put a widget on our website which always displays an up-to-date estimate of the current actual price of running various transactions on our stack. Any measurement which doesn't allow us to directly derive those values is, by comparison, weak information.

@jacque006 Just acknowledging here that we did discuss adding this kind of information as followup. Can we link that here when it's created? I just can't stress enough that these measurements are an incomplete picture without that next step.

**Comment by jacque006** [comment]
> Maybe the value is in refining/practicing the process before spending the fees on mainnet?

This is the primary reason I started on a testnet. Also, the analysis of those testnet values & the presentation of such will likely be the same across network runs, so would like to refine that before hitting other networks/mainnets.

> 
> Similar to what we touched on in our call, the (L1 gas, L2 gas) pair is the thing that we can control. If we don't have those two numbers (remembering that "gas used for L1" is not L1 gas), then the measurement we have will be subject to the constant gas variability that is outside of our project.
> 
> To put it another way, if we have the (L1 gas, L2 gas) pair, then we could put a widget on our website which always displays an up-to-date estimate of the current actual price of running various transactions on our stack. Any measurement which doesn't allow us to directly derive those values is, by comparison, weak information.

I agree that that L1 block, gas, and price information is valuable for us & other L2 devs and I would like to include it eventually on all runs. However, I think to start most people are going to mainly care about the actual ETH/USD on that L2 and the difference if a BLS bundle vs. a normal txn, such as what is shown on https://l2fees.info/ .

> @jacque006 Just acknowledging here that we did discuss adding this kind of information as followup. Can we link that here when it's created? I just can't stress enough that these measurements are an incomplete picture without that next step.

Yep, will do. It will be part of https://github.com/web3well/bls-wallet/milestone/9

**Comment by voltrevo** [comment]
> eventually

Shouldn't it be the very next thing?

I feel like we're on the same page about everything now except for the importance of this extra information. Alternatively - Is there a difficulty in this step that motivates 'eventually' that I'm not seeing?

I'm not sure what the disconnect is here. To me it is very clear that it would be grossly incomplete to not include L1 gas in the measurements. l2fees.info doesn't have the same problem because it updates in real-time. Without real-time updates, reporting the ETH cost, or for this case even reporting gas, for the same reason, is going to be frequently wrong and its usefulness therefore severely limited.

Even more important than reporting these numbers externally is our ability to measure our progress and make internal decisions. For that use case, using this fluctuating measurement is even more of a problem.

For example - right now mainnet is charging 40+ gwei for gas, which is something like 4x what it was yesterday. This means if we measured arbitrum (real) right now without accounting for this, the 'gas used for L1' is going to be 4x higher than normal. I just don't see why that problem is something we can say 'eventually' about.
